inst/cpp/sim.cpp

```diff
@@ -63,23 +63,116 @@ void pk_code (int i, std::vector<double> times, std::vector<double> doses, doubl
   // insert custom pk event code
 }
 
+NumericVector lagtime_to_numeric(SEXP lagtime, List parameters) {
+   NumericVector lagtime_numeric;
+  if (TYPEOF(lagtime) == REALSXP) {
+    lagtime_numeric = as<NumericVector>(lagtime);
+  } else if (TYPEOF(lagtime) == STRSXP) {
+    CharacterVector lagtime_char = as<CharacterVector>(lagtime);
+    lagtime_numeric = NumericVector(lagtime_char.size());
+    for (int i = 0; i < lagtime_char.size(); i++) {
+      String param_name = lagtime_char[i];
+      if (parameters.containsElementNamed(param_name.get_cstring())) {
+        lagtime_numeric[i] = as<double>(parameters[param_name]);
+      } else {
+        lagtime_numeric[i] = 0.0; // default value
+      }
```

Comment:
am i correct that this means that if we supply lagtime as a variable (e.g. "ALAG") but the supplied parameters do not contain that variable, we quietly override with zero? should we implement a check for this on the R side maybe?

Reply:
yes, that's how it works. Good point, I'll add a warning on R side